### PR TITLE
Suppression support for `text-matches`

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: src/findReadMe.ts [hunk 1]
```diff
@@ -19,6 +19,14 @@ export const findReadMe = async (dir: string): Promise<string | undefined> => {
         if (newDir === dir) {
             return undefined
         }
+        const url = vfs.urlParse(newDir)
+        if (url !== undefined) {
+            const pathSplit = url.path.split("/")
+            if (pathSplit.length === 2 && pathSplit[0] === "github.com") {
+                // return undefined if it's a GitHub organization instead of a repository
+                return undefined
```
Comment (billytrend) [inline src/findReadMe.ts]: can we log here or is logging not available?

File: src/readMeManipulator.ts [hunk 2]
```diff
@@ -81,26 +84,28 @@ export class ReadMeManipulator {
...
-    public getTagsForFilesChanged(markDownEx: MarkDownEx, specsChanged: ReadonlyArray<string>): ReadonlyArray<string> {
+    public readonly getTagsForFilesChanged = (
+        markDownEx: MarkDownEx,
+        specsChanged: ReadonlyArray<string>
+    ): ReadonlyArray<string> => {
         const codeBlocks = getTagsToSettingsMapping(markDownEx.markDown);
         const tagsAffected = new Set<string>();
 
-        for (const [tag, settings] of Object.entries(codeBlocks)) {
+        for (const [tag, settings] of sm.entries(codeBlocks)) {
```
Comment (billytrend) [inline src/readMeManipulator.ts]: nooo...whyyy

File: src/readMeManipulator.ts [hunk 3]
```diff
@@ -184,8 +192,9 @@ export interface CodeBlocksAndHeadings {
 
 export const getCodeBlocksAndHeadings = (
     startNode: commonmark.Node
-): CodeBlocksAndHeadings => {
-    return getAllCodeBlockNodes(startNode).reduce(
+): CodeBlocksAndHeadings =>
+    it.fold(
```
Comment (billytrend) [inline src/readMeManipulator.ts]: 👎

File: src/readMeManipulator.ts [hunk 4]
```diff
@@ -214,18 +222,17 @@ const getHeadingLiteral = (heading: commonmark.Node): string => {
...
 
-const getAllCodeBlockNodes = (startNode: commonmark.Node) => {
-    const walker = startNode.walker()
-    const result: commonmark.Node[] = []
-    while (true) {
-        const a = walkToNode(walker, n => n.type === "code_block")
-        if (!a) {
-            break
+const getAllCodeBlockNodes = (startNode: commonmark.Node) =>
+    it.iterable(function *() {
```
Comment (billytrend) [inline src/readMeManipulator.ts]: What is the value add of these changes? I can't see any advantage myself

Comment (billytrend) [reply src/readMeManipulator.ts]: making a generator is nice but cant we just use vanilla javascript

File: src/findReadMe.ts (hunk 1 above)
Comment (sergey-shandar) [reply src/findReadMe.ts]: It's not available at this point. We need to think about solution here.

**Comment by sergey-shandar** [comment]
@fearthecowboy as discussed, the new field `text-matches` in suppressions.

File: src/findReadMe.ts (hunk 1 above)
Comment (sergey-shandar) [reply src/findReadMe.ts]: As discussed, we may have another library `@ts-common/log` which will define common interface and provide logger instance. The library should wrap one of the existing library.

**Pull request merged**